### PR TITLE
[ISSUE #102] 修复 circuitbreaker 熔断指标上报 prometheus 出现 nil 导致 panic

### LINKED ISSUE #102: 运行polaris-go里的example的circuitbreaker时崩溃报空指针错误
**Describe the bug**
运行polaris-go里的example的circuitbreaker时崩溃报空指针错误

**To Reproduce**
分别运行circuitbreaker的provider 和 consumer ，运行两个provider实例，按照readme步骤测试熔断功能，先停止一个provider实例，curl 向 consumer再发起请求，consumer就崩溃了。

**Expected behavior**
希望能正常测试熔断功能

**Environment**

- Version: [master]
- OS: [centos 7.9]

**Additional context**
分别运行circuitbreaker的provider 和 consumer ，运行两个provider实例，按照readme步骤测试熔断功能，先停止一个provider实例，curl 向 consumer再发起请求，consumer就崩溃了。

### PR DESCRIPTION
**Pull request by chuntaojun**
**Please provide issue(s) of this PR:**
Fixes #102 

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration
- [ ] Docs
- [ ] Performance and Scalability
- [x] Naming
- [ ] HealthCheck
- [ ] Test and Release

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any user-facing changes. This may include API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: plugin/statreporter/prometheus/prometheus_handler.go [hunk 1]
```diff
@@ -122,6 +123,13 @@ func (p *PrometheusHandler) registerMetrics() error {
 
 // ReportStat 上报采集指标到 prometheus，这里只针对部分 model.InstanceGauge 的实现做处理
 func (p *PrometheusHandler) ReportStat(metricsType model.MetricType, metricsVal model.InstanceGauge) error {
+	defer func() {
```
Comment (andrewshan) [inline plugin/statreporter/prometheus/prometheus_handler.go:126]: 这里不需要加recover吧？我们自己代码要保证逻辑不会panic

File: plugin/statreporter/prometheus/prometheus_handler.go [hunk 2]
```diff
@@ -223,21 +231,22 @@ func (p *PrometheusHandler) handleRateLimitGauge(metricsType model.MetricType, v
...
+	open := p.metricVecCaches[MetricsNameCircuitBreakerOpen].(*prometheus.GaugeVec)
 
-	status := val.GetCircuitBreakerStatus().GetStatus()
+	// 计算完之后的熔断状态
+	status := val.GetNextCircuitBreakerStatus().GetStatus()
 	if status == model.Open {
 		open.With(labels).Inc()
 	} else {
-		open.With(labels).Add(-1)
+		open.With(labels).Dec()
```
Comment (andrewshan) [inline plugin/statreporter/prometheus/prometheus_handler.go:241]: 这2个方法有什么区别吗？

File: pkg/model/service.go [hunk 3]
```diff
@@ -772,12 +772,18 @@ type CircuitBreakGauge struct {
 	EmptyInstanceGauge
 	ChangeInstance Instance
 	Method         string
-	CBStatus       CircuitBreakerStatus
+	PreCBStatus    CircuitBreakerStatus
```
Comment (andrewshan) [inline pkg/model/service.go:775]: 新加这个属性是做什么用，好像后面并没有用到？

File: plugin/statreporter/prometheus/prometheus_handler.go (hunk 2 above)
Comment (chuntaojun) [reply plugin/statreporter/prometheus/prometheus_handler.go:233]: Add 只能非负数，Dec 才能用于指标值减少

**Review by houseme — APPROVED** [review]

**Pull request merged**